### PR TITLE
ENH: Add tolerance param to np.random.multinomial

### PR DESCRIPTION
**Pull request by alexbeloi**
Currently `multinomial(n, pvals, size=None)`  checks/raises exception if `sum(pvals) > 1 + 1e-12`, I have found this tolerance to be too small in some cases. 

Specifically when interacting with tensorflow, `tf.nn.softmax` outputs a tensor whose values should sum to 1, but often only within `1e-8`. 

In this patch:
- added parameter `tol=1e-12`
- changed exception to `sum(pvals) > 1 + tol`
- added exception if `tol < 0`
- added documentation for 'Raises'

### CODE REVIEW

**Comment by alexbeloi** [comment]
Found workaround, closing this PR with comment for anyone else that has this issue.

On the numpy mailing list, @shoyer mentioned that tensorflow has its own implementation tf.multinomial which takes logits (i.e. renormalizes probability values before sampling) this works perfectly for my use case.
